diff --git a/sites/all/modules/wmf_civicrm/db_switcher.inc b/sites/all/modules/wmf_civicrm/db_switcher.inc
index 19e5b35..b8999fa 100644
--- a/sites/all/modules/wmf_civicrm/db_switcher.inc
+++ b/sites/all/modules/wmf_civicrm/db_switcher.inc
@@ -57,17 +57,21 @@ class db_switcher {
   }
 
   static function parse_dbi_url( $url ) {
-    if (!preg_match( '/^([a-z]+):\/\/([^:]+):([^@]+)@([^\/]+)\/(.+)$/', $url, $matches )) {
+    if (!preg_match( '/^([a-z]+):\/\/([^:]+):([^@]+)@([^\/:]+)(:([0-9]+))?\/(.+)$/', $url, $matches )) {
         throw new Exception("Failed to parse dbi url: $url");
     }
 
-    return array(
+    $r = array(
         'driver' => str_replace( 'mysqli', 'mysql', $matches[1] ),
         'username' => $matches[2],
         'password' => $matches[3],
         'host' => $matches[4],
-        'database' => $matches[5],
+        'database' => $matches[7],
     );
+    if ($matches[6]) {
+        $r['port'] = $matches[6];
+    }
+    return $r;
   }
     
   /**
