diff --git a/includes/database/mysql/database.inc b/includes/database/mysql/database.inc
index 356e039f..c803207d 100644
--- a/includes/database/mysql/database.inc
+++ b/includes/database/mysql/database.inc
@@ -101,6 +101,14 @@ class DatabaseConnection_mysql extends DatabaseConnection {
     }
   }
 
+  public function query($query, array $args = array(), $options = array()) {
+    $query = preg_replace('/{([^}]+)}/', '`\1`', $query);
+    // This to make Drush work
+    $query = str_replace(' system.', ' `system`.', $query);
+    $query = str_replace('`system` system', '`system` `system`', $query);
+    return parent::query($query, $args, $options);
+  }
+
   public function queryRange($query, $from, $count, array $args = array(), array $options = array()) {
     return $this->query($query . ' LIMIT ' . (int) $from . ', ' . (int) $count, $args, $options);
   }
