commit 92e2c12c83bba39f11c2515ef7809e777d3e05b4
Author: Tim Otten <totten@civicrm.org>
Date:   Thu Jan 22 23:28:32 2026 -0800

    settings.php - Support db.port

diff --git a/drupal/sites/default/settings.php b/drupal/sites/default/settings.php
index 6e235bec0..658e16e18 100644
--- a/drupal/sites/default/settings.php
+++ b/drupal/sites/default/settings.php
@@ -47,7 +47,7 @@ $databases = array (
       'username' => $bbconfig['db.user'],
       'password' => $bbconfig['db.pass'],
       'host'     => $bbconfig['db.host'],
-      'port'     => '',
+      'port'     => $bbconfig['db.port'] ?? '',
       'prefix'   => '',
     ),
   ),
