commit 3fa9736046e615053ab3be3042f03d904e9a8e82
Author: Tim Otten <totten@civicrm.org>
Date:   Wed Jan 28 15:54:49 2026 -0800

    civicrm.settings.php - Accept http.port option

diff --git a/drupal/sites/default/civicrm.settings.php b/drupal/sites/default/civicrm.settings.php
index 848f9c898..569a4dd6f 100644
--- a/drupal/sites/default/civicrm.settings.php
+++ b/drupal/sites/default/civicrm.settings.php
@@ -33,6 +33,7 @@ $shortname = get_config_value($bbcfg, 'shortname', null);
 $approot = get_config_value($bbcfg, 'app.rootdir', null);
 $drupalroot = get_config_value($bbcfg, 'drupal.rootdir', null);
 $dataroot = get_config_value($bbcfg, 'data.rootdir', null);
+$httpPort = get_config_value($bbcfg, 'http.port', null);
 
 if (!$servername || !$shortname || !$approot || !$drupalroot || !$dataroot) {
   die("Incorrect config; check these settings: servername, shortname, app.rootdir, drupal.rootdir, data.rootdir\n");
@@ -55,7 +56,12 @@ define('CIVICRM_UF_DSN', $bbcfg['drupal_db_url'].'?new_link=true');
 define('CIVICRM_LOGGING_DSN', $bbcfg['log_db_url'].'?new_link=true');
 
 define('CIVICRM_TEMPLATE_COMPILEDIR', "$dataroot/$datadirname/civicrm/templates_c");
-define('CIVICRM_UF_BASEURL', "http://$servername/");
+define('CIVICRM_UF_BASEURL', implode('', [
+  'http://',
+  $servername,
+  empty($httpPort) ? '' : (':' . $httpPort),
+  '/',
+]));
 define('CIVICRM_SITE_KEY', get_config_value($bbcfg, 'site.key', '32425kj24h5kjh24542kjh524'));
 
 define('CIVICRM_DOMAIN_ID', 1);
