diff --git a/CRM/Core/ClassLoader.php b/CRM/Core/ClassLoader.php
index 9af4ce3238..97cc27ba4c 100644
--- a/CRM/Core/ClassLoader.php
+++ b/CRM/Core/ClassLoader.php
@@ -124,7 +124,12 @@ class CRM_Core_ClassLoader {
     $this->initHtmlPurifier($prepend);
 
     $this->_registered = TRUE;
-    $packages_path = implode(DIRECTORY_SEPARATOR, [$civicrm_base_path, 'packages']);
+    if (isset($GLOBALS['civicrm_paths']['civicrm.packages']['path'])) {
+      $packages_path = $GLOBALS['civicrm_paths']['civicrm.packages']['path'];
+    }
+    else {
+      $packages_path = implode(DIRECTORY_SEPARATOR, [$civicrm_base_path, 'packages']);
+    }
     $include_paths = [
       '.',
       $civicrm_base_path,
